tests/phpunit/tests/block-supports/elements.php

```diff
@@ -3,171 +3,306 @@
  * @group block-supports
  *
  * @covers ::wp_render_elements_support
+ * @covers ::wp_render_elements_support_styles
```

Comment:
Test classes should only have `@covers` annotations if all tests cover all of the functions listed in the annotation.

There's been an effort recently to follow a pattern of separating tests for a given function into their own file, so I'd recommend putting the `wp_render_elements_support_styles()` tests in their own file and class:

File name: `wpRenderElementsSupportStyles.php`
Class name: `Tests_Block_Supports_WpRenderELementsSupportStyles`
Covers annotation: `@covers ::wp_render_elements_support_styles`

Reply:
+1 for removing the `@covers` annotations from the class and putting on the individual functions.

However, I'm not sure about moving to function-specific files. @costdev Can you point to where this was decided? Or is being discussed? I have never heard about this guidance before, so I want to make sure we're not making changes to the unit tests structure preliminarily.

Reply:
@felixarntz It's not that it was decided as a change - There's a bit of a mixed history in the Core test suite of having one file per function, or dumping everything into one place (e.g. option.php for one we've seen recently).

The [Writing PHPUnit Tests Handbook entry](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes) gives an example of a file that tests the `get_comment_class()` function, with its own file and class. The handbook entry doesn't impose any restrictions, but as this file was previously only testing one function, and will now be testing two, I thought I'd suggest creating a separate file.

The effort I'm referring to is just that several Gutenberg test files have been split into per-function files, and while this divides up a given feature's tests (which would otherwise be put into one file), it isolates a given function's tests.

Reply:
Actually, allow me to correct myself:

The handbook entry actually says:
> As a rule, a single test class should not contain tests for more than one function/method.
> [Reference](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes)

Noting that it says _should not_, rather than _must not_, so there's room for nuance.

Reply:
Thanks for the feedback 👍 

I've split the tests into separate files per function as per this [comment](https://github.com/WordPress/wordpress-develop/pull/5418#discussion_r1348478534). See https://github.com/WordPress/wordpress-develop/pull/5418/commits/a7dbbb7d5c08775aaab81a414d6977137bfef605